src/v/cluster/scheduling/leader_balancer_greedy.h

```diff
@@ -53,20 +60,18 @@ class greedy_balanced_shards final : private leader_balancer_strategy {
     /*
      * Compute new error for the given reassignment.
      */
-    double adjusted_error(
+    double adjusted_error(double current_error,
```

Comment:
By passing the error in here, we don't need to calculate it every time: since `error()` involves a loop over _every_ shard every time it is called, it was taking more than half of the total time.

Reply:
🤯😎